### PR TITLE
Add method for GET /api/cloud_user

### PR DESCRIPTION
**Pull request by velovix**
Adds a method for an API that allows clients to get information on the currently logged-in cloud user. The server returns a 404 when a user is not logged in, but this method returns `None` in this case instead.

### CODE REVIEW

**Comment by BryceBeagle** [comment]
It might be better to call this `get_current_cloud_user` or something? Seeing just the endpoint makes me think it should be something that takes a user_id as an argument and returns info about that user.

**Review by apockill — APPROVED** [review]

**Comment by velovix** [comment]
@BryceBeagle Sounds reasonable

File: brainframe/api/stubs/cloud_users.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+        :param timeout: The timeout to use for this request
+        :return: Information on the cloud user, or None if no user is logged in
+        """
+        req = f"/api/cloud_user"
+        try:
+            data, _ = self._get_json(req, timeout=timeout)
+            return CloudUserInfo.from_dict(data)
+        except CloudUserNotFoundError:
+            return None
```
Comment (BryceBeagle) [inline brainframe/api/stubs/cloud_users.py]: Nitpicky change. `.from_dict` can't raise a `CloudUserNotFoundError`

```suggestion
        try:
            data, _ = self._get_json(req, timeout=timeout)
        except CloudUserNotFoundError:
            return None
        else:
            return CloudUserInfo.from_dict(data)
```

**Review by BryceBeagle — CHANGES_REQUESTED** [review]

File: brainframe/api/stubs/cloud_users.py (hunk 1 above)
Comment (velovix) [reply brainframe/api/stubs/cloud_users.py]: Good point

**Review by BryceBeagle — APPROVED** [review]

**Pull request merged**